lms/djangoapps/courseware/module_render.py

```diff
@@ -772,7 +785,9 @@ def xqueue_callback(request, course_id, userid, mod_id, dispatch):
     if not isinstance(header, dict) or 'lms_key' not in header:
         raise Http404
 
-    instance = load_single_xblock(request, userid, course_id, mod_id)
+    course = get_course_by_id(course_id)
+
+    instance = load_single_xblock(request, userid, course_id, mod_id, course=course)
```

Comment:
This should be wrapped in `modulestore().bulk_operation(course_id)`, so that loading the course and then loading a single XBlock doesn't encur a double database hit.